x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/utils/indexing_service.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getHttp } from '../../../../kibana_services';
+import { CreateDocSourceResp, INDEX_SOURCE_API_PATH } from '../../../../../common';
+
+export const createNewIndexAndPattern = async (indexName: string) => {
+  return await getHttp().fetch<CreateDocSourceResp>({
+    path: `/${INDEX_SOURCE_API_PATH}`,
+    method: 'POST',
+    body: convertObjectToBlob({
```

Comment:
Why is the body getting converted to a blob? Why not just pass json to the API?

Reply:
this is because of the type definition https://github.com/elastic/kibana/blob/7ede7a8eca815db738e53e3b5ee51d9daa633732/src/core/public/http/types.ts#L158

`BodyInit` needs to be a blob.

Reply:
I don't think you need to blob. Checkout out the calls in https://github.com/elastic/kibana/blob/master/x-pack/plugins/file_upload/public/api/index.ts#L89 and https://github.com/elastic/kibana/blob/master/x-pack/plugins/file_upload/public/importer/importer.ts#L235. Looks like you could just convert to a string.